app/furniture/marketplace/orders_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Marketplace
+  class OrdersController < Controller
+    def show
+      authorize(order)
+    end
+
+    helper_method def order
+      policy_scope(marketplace.orders).find(params[:id])
+    end
+  end
+end
```

Comment:
I am trying to visualize the way all these data models interact as the user goes thru the shopping flow is. Is it like this?

1. A customer adds items to their cart == a `Cart` of `pre_checkout` status contains `cart_products`.
2. The customer goes to the checkout page and is filling out credit card info to complete their purchase ==  a `Cart` of `pre_checkout` status is going thru the stripe checkout flow thus creating a `Checkout` record. 
3. if the checkout flow via stripe is successful, the `Cart` changes to `check_out` status, and an `Order` is created from the `cart_products`.

Reply:
So, let's decompose the datastructure a bit here:

1. at the *database* layer, a `Checkout` and  an `Order` are equivalent; they both store their attributes in the same table.
2. At the application layer, a `Checkout` and an `Order` have different routes, controllers, policies, and views.
3. At present, whether something is represented as a `Checkout` or an `Order` depends on what url is being used to retrieve the data. If the request path includes `marketplace/*/checkouts/*` then the data stored in the database will be treated as a `Checkout`. If it's `marketplace/*/orders` then the data stored in the database will be treated as an `Order`.

So, an `Order` is not 'created' in the sense that it's not actually creating any records in the database table; rather we leverage "polymorphism" (which can be loosely defined as 'the ability for one data structure to be treated as another data structure') so that we have more Domain objects for different use cases; while retaining a more normalized database.

It's entirely possible that we *do* want an `Order` to be a distinct database table from a `Checkout`, or we may not want a `Checkout` to not be a database-backed model at all! 

So, to reframe:

1. `Shopper` adds `Product`s to their `Cart`, creating a `CartProduct` record which keeps track of things like quantity. (note: I think a `Cart` is also a kind of `Order` or `Checkout`, which I think @anaulin may agree with)
2. `Shopper` creates a `Checkout`, which directs them to Stripe to collect payment  information.
3. `Shopper` completes `Checkout`, which directs them back from Stripe to Convene, and updates the `Checkout` object to indicate it had been succesfully checked-out.
4. `Shopper` is redirected to show the `Order`, which represents the `Cart` to `Shopper`s in a read-only manner.

Reply:
wow this is great! once Marketplace is finished/finalized/stable we could make a seperate .MD doc readme with the above info and maybe a diagram ❤️

Reply:
I think it could be dropped directling into `app/furniture/marketplace/README.md` in it's own PR :D